src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshalling/Utf16StringMarshaller.cs

```diff
@@ -9,64 +9,29 @@ namespace System.Runtime.InteropServices.Marshalling
     /// Marshaller for UTF-16 strings
     /// </summary>
     [CLSCompliant(false)]
-    [CustomTypeMarshaller(typeof(string), BufferSize = 0x100,
-        Features = CustomTypeMarshallerFeatures.UnmanagedResources | CustomTypeMarshallerFeatures.TwoStageMarshalling | CustomTypeMarshallerFeatures.CallerAllocatedBuffer)]
+    [CustomTypeMarshaller(typeof(string),
+        Features = CustomTypeMarshallerFeatures.UnmanagedResources | CustomTypeMarshallerFeatures.TwoStageMarshalling)]
     public unsafe ref struct Utf16StringMarshaller
     {
-        private ushort* _allocated;
-        private readonly Span<ushort> _span;
+        private ushort* _nativeValue;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="Utf16StringMarshaller"/>.
         /// </summary>
         /// <param name="str">The string to marshal.</param>
```

Comment:
```suggestion
        /// <remarks>
        /// The caller allocated constructor option is not provided because
        /// pinning should be preferred for UTF-16 scenarios.
        /// </remarks>
        /// <param name="str">The string to marshal.</param>
```